ec2api/api/ec2utils.py

```diff
@@ -223,6 +223,44 @@ def get_db_item(context, ec2_id, expected_kind=None):
     return item
 
 
+def get_db_item_cross_account(context, ec2_id, expected_kind=None):
+    """Get an DB item, raise AWS compliant exception if it's not found.
+
+        Args:
+            context (RequestContext): The request context.
+            ec2_id (str): The ID of the requested item.
+            expected_kind (str): The expected kind of the requested item.
+                It should be specified for a kind of ec2_id to be validated,
+                if you need it.
+
+        Returns:
+            The DB item.
+    """
+    item = db_api.get_item_by_id_cross_account(context, ec2_id)
+    if (item is None or
```

Comment:
That is correct ! But it will still raise exception for InvalidSubnetIDNotFound

`raise NOT_FOUND_EXCEPTION_MAP[kind](**params)`
However code at 
https://github.com/hmdesai89/ec2-api/blob/393d200071a0c73bdba3bb02ae3f4fbd657368df/ec2api/api/network_interface.py#L57
Will never go in if condition